lib/util/bignumber.js

```diff
@@ -281,6 +281,60 @@ exports.or = function (x, y) {
   return bitwise(x, y, function (a, b) { return a | b });
 };
 
+/**
+ * Calculate the sine of x using Taylor Series.
+ *
+ * sin(x) = x - x^3/3! + x^5/5! - x^7/7! + x^9/9! - ...
+ *        = x - x^2*x^1/3! + x^2*x^3/5! - x^2*x^5/7! + x^2*x^7/9! - ...
+ *
+ * TODO: Replace with Chebyshev approximation.
+ *
+ * @param {BigNumber} x
+ * @returns {BigNumber} sine of x
+ */
+exports.sin = function (x) {
+  var BigNumber = x['constructor'];
+  var precision = BigNumber['precision'];
+  if (x.isNaN() || !x.isFinite()) {
+    return new BigNumber(NaN);
+  }
+
+  // Get number's offset within the period of sin (tau)
+  var pi = exports.pi(2 * precision - 5);
+  var tau = pi.times(2);
+
+  // Catch if tau multiple using pi's precision
+  if (x.div(pi.toDP(x.dp(), 1)).toNumber() % 2 == 0) {
+    return new BigNumber(0);
+  }
+
+  var y = x.mod(tau);
+  // Catch if tau multiple with tau's precision
+  if (y.toDP(x.dp(), 1).isZero()) {
+    return new BigNumber(0);
+  }
+  if (y.gt(pi)) {
+    y = y.minus(tau);
+  }
+  y['constructor']['precision'] = precision;
```

Comment:
Just that this isn't the official way of configuring the precision, which is the [`config({...})`](http://mikemcl.github.io/decimal.js/#Dconfig) function. Anyway it's fine with me, I'm quite sure our unit tests will neatly break if some day the internals o decimal.js change.